analysis/README.md

```diff
@@ -0,0 +1,623 @@
+pol commit reg_write_Y_A;
+pol commit A;
+pol commit instr_identity;
+pol commit instr_one;
+pol commit instr_nothing;
+pol commit instr__jump_to_operation;
+pol commit instr__reset;
+pol commit instr__loop;
+pol commit instr_return;
+pol commit X_const;
+pol commit X_read_free;
+pol commit read_X_A;
+pol commit read_X_pc;
+X = ((((read_X_A * A) + (read_X_pc * pc)) + X_const) + (X_read_free * X_free_value));
+pol commit Y_const;
+pol commit Y_read_free;
+pol commit read_Y_A;
+pol commit read_Y_pc;
+Y = ((((read_Y_A * A) + (read_Y_pc * pc)) + Y_const) + (Y_read_free * Y_free_value));
+pol constant first_step = [1] + [0]*;
+A' = ((((reg_write_X_A * X) + (reg_write_Y_A * Y)) + (instr__reset * 0)) + ((1 - ((reg_write_X_A + reg_write_Y_A) + instr__reset)) * A));
+pc' = ((1 - first_step') * ((((instr__jump_to_operation * _function_id) + (instr__loop * pc)) + (instr_return * 0)) + ((1 - ((instr__jump_to_operation + instr__loop) + instr_return)) * (pc + 1))));
+pol constant p_line = [0, 1, 2, 3, 4] + [4]*;
+pol commit X_free_value(i) query match pc {  };
+pol commit Y_free_value(i) query match pc {  };
+pol constant p_X_const = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_X_read_free = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_Y_const = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_Y_read_free = [0, 0, 1, 0, 0] + [0]*;
+pol constant p_instr__jump_to_operation = [0, 1, 0, 0, 0] + [0]*;
+pol constant p_instr__loop = [0, 0, 0, 0, 1] + [1]*;
+pol constant p_instr__reset = [1, 0, 0, 0, 0] + [0]*;
+pol constant p_instr_identity = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_instr_nothing = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_instr_one = [0, 0, 1, 0, 0] + [0]*;
+pol constant p_instr_return = [0, 0, 0, 1, 0] + [0]*;
+pol constant p_read_X_A = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_read_X_pc = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_read_Y_A = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_read_Y_pc = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_reg_write_X_A = [0, 0, 0, 0, 0] + [0]*;
+pol constant p_reg_write_Y_A = [0, 0, 1, 0, 0] + [0]*;
+{ pc, reg_write_X_A, reg_write_Y_A, instr_identity, instr_one, instr_nothing, instr__jump_to_operation, instr__reset, instr__loop, instr_return, X_const, X_read_free, read_X_A, read_X_pc, Y_const, Y_read_free, read_Y_A, read_Y_pc } in { p_line, p_reg_write_X_A, p_reg_write_Y_A, p_instr_identity, p_instr_one, p_instr_nothing, p_instr__jump_to_operation, p_instr__reset, p_instr__loop, p_instr_return, p_X_const, p_X_read_free, p_read_X_A, p_read_X_pc, p_Y_const, p_Y_read_free, p_read_Y_A, p_read_Y_pc };
+pol constant _block_enforcer_last_step = [0]* + [1];
+pol commit _function_id_no_change;
+_function_id_no_change = ((1 - _block_enforcer_last_step) * (1 - instr_return));
+(_function_id_no_change * (_function_id' - _function_id)) = 0;
+// Links:
+// // // instr identity with params  X -> Y links to // // function identity with id 2 with params  _input_0 -> _output_0 in object at location main_sub with latch "instr_return" and function_id "_function_id"
```

Comment:
Not too important but redundant `//` here and later in the same line

Reply:
also, where are these links?

Reply:
Fixed. What do you mean where? They belong to the object.

Reply:
yes but at first I thought the comments were referring to something right underneath or something close